pkg/snapshot/restorer/restorer.go

```diff
@@ -796,31 +796,32 @@ func verifySnapshotRevision(clientKV client.KVCloser, snap *brtypes.Snapshot) er
 	return nil
 }
 
-// getDecompressedSnapshotReadCloser passes the given ReadCloser through the
+// getNormalizedSnapshotReadCloser passes the given ReadCloser through the
 // snapshot decompressor if the snapshot is compressed using a compression policy.
-// Also returns whether the snapshot was initially compressed or not, as well as
+// If snapshot is not compressed, it returns the given ReadCloser as is.
+// It also returns whether the snapshot was initially compressed or not, as well as
 // the compression policy used for compressing the snapshot.
-func getDecompressedSnapshotReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) (io.ReadCloser, bool, string, error) {
+func getNormalizedSnapshotReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) (io.ReadCloser, bool, string, error) {
 	isCompressed, compressionPolicy, err := compressor.IsSnapshotCompressed(snap.CompressionSuffix)
 	if err != nil {
-		return nil, false, compressionPolicy, err
+		return rc, false, "", err
 	}
 
 	if isCompressed {
 		// decompress the snapshot
 		rc, err = compressor.DecompressSnapshot(rc, compressionPolicy)
 		if err != nil {
-			return nil, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
+			return rc, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
```

Comment:
Do we want to return `rc` here even though `compressor.DecompressSnapshot` returned an error?

Reply:
Yes. Caller should still be able to use the original `rc` is they wish to, for whatever reason it may be. In any case, in the current PR code, the returned `rc` is never used if there's an error returned.